src/Framework/Constraint/JsonMatchesErrorMessageProvider.php

```diff
@@ -17,7 +17,7 @@ class JsonMatchesErrorMessageProvider
     /**
      * Translates JSON error to a human readable string.
      */
-    public static function determineJsonError(string $error, string $prefix = ''): ?string
+    public static function determineJsonError(int $error, string $prefix = ''): ?string
```

Comment:
Perhaps, but the signature is obviously incorrect, and would cause a `TypeError` when called if `declare(strict_types=1)` were used. It was added in https://github.com/sebastianbergmann/phpunit/commit/5113b86ce113e531b65b1d5563d0052461bb2ee6#diff-fa44678565c7f24dfaaa94a897531956R21

Reply:
yet, strict types are not in use for this file.

i'm not saying that this change is bad.
I'm saying this change is BC breaker

Reply:
Ah right, ok. Does BC breaker mean a new major version?